From ad6b6e42a2f18e812cb7ece65d46f8f9504a66b9 Mon Sep 17 00:00:00 2001
From: zain wang <wzz@rock-chips.com>
Date: Mon, 11 Jun 2018 11:04:25 +0800
Subject: [PATCH] mfd: fusb302: Don't ignore UFP_D

Don't ignore UFP_D data, so that we can get
right DP pin cap from DISCOVERY_MODES.

Fix commit 4258959427ec3c211f9d99e4e2713385b16cb8ee
mfd: fusb302: add DP UFP_D support

Change-Id: I6a6250d26c1a847c9b75c676b14f47a37bfcc1f9
Signed-off-by: zain wang <wzz@rock-chips.com>
---
 drivers/mfd/fusb302.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/fusb302.c b/drivers/mfd/fusb302.c
index 39f26e833385..f60375e159a4 100644
--- a/drivers/mfd/fusb302.c
+++ b/drivers/mfd/fusb302.c
@@ -1103,11 +1103,12 @@ static void process_vdm_msg(struct fusb30x_chip *chip)
 				 * store mode config,
 				 * enter first mode default
 				 */
-				if (!((chip->rec_load[1] >> 8) & 0x3f)) {
+				tmp = chip->rec_load[1];
+				if ((!((tmp >> 8) & 0x3f)) &&
+				    (!((tmp >> 16) & 0x3f))) {
 					chip->val_tmp |= 1;
 					break;
 				}
-				tmp = chip->rec_load[1];
 				chip->notify.pin_assignment_def = 0;
 				chip->notify.pin_assignment_support =
 					VDM_DP_GET_RECEPTACLE(tmp) ?
-- 
2.35.3

